docs/build/autogenerate.rst

```diff
@@ -132,10 +132,12 @@ Autogenerate can **optionally detect**:
 * Change of column type.  This will occur if you set
   the :paramref:`.EnvironmentContext.configure.compare_type` parameter
   to ``True``, or to a custom callable function.   The default implementation
-  **only detects major type changes**, such as between ``Numeric`` and
-  ``String``, and does not detect changes in arguments such as lengths, precisions,
-  or enumeration members.  The type comparison logic is extensible to work
-  around these limitations, see :ref:`compare_types` for details.
+  will reliably detect major changes, such as between ``Numeric`` and
+  ``String``.  If both the existing type and the new type have keyword arguments
+  specified (such as lengths, precisions, or enumeration members), they will be
+  compared as well.  However, if keywords are only specified for one,
+  changes in these will be ignored. The type comparison logic is extensible to
+  work around these limitations, see :ref:`compare_types` for details.
```

Comment:
**mike bayer** (zzzeek) wrote:

add:

  .. versionchanged:: 1.4   type comparison code has been enhanced to compare column types more deeply as well as to take arguments into account.